app/assets/javascripts/dragdrop.js

```diff
@@ -9,216 +9,132 @@ jQuery(document).ready(function() {
  * #side-dropzone, is for the main image of research notes, in /app/views/editor/post.html.erb
 */
 
-    function progressAll(elem, data) {
-        var progress = parseInt(data.loaded / data.total * 100, 10);
-        $(elem).css(
-            'width',
-            progress + '%'
-        );
-    }
+  function progressAll(elem, data) {
+    var progress = parseInt(data.loaded / data.total * 100, 10);
+    $(elem).css(
+        'width',
+        progress + '%'
+    );
+  }
 
-    $('.dropzone').bind('dragover',function(e) {
-      e.preventDefault();
-      $('.dropzone').addClass('hover');
-    });
-    $('#side-dropzone').bind('dragover',function(e) {
-      e.preventDefault();
-      $('#side-dropzone').addClass('hover');
-    });
-    $('.dropzone').bind('dragout',function(e) {
-      $('.dropzone').removeClass('hover');
-    });
-    $('#side-dropzone').bind('dragout',function(e) {
-      $('#side-dropzone').removeClass('hover');
-    });
-    $('.dropzone').bind('drop',function(e) {
-      $D.selected = $(e.target.closest('div.comment-form-wrapper')).eq(0);
-      e.preventDefault();
-      $E.initialize({});
-    });
-    $('#side-dropzone').bind('drop',function(e) {
-      e.preventDefault();
-    });
+  $('.dropzone').on('dragover',function(e) {
+    e.preventDefault();
+    $('.dropzone').addClass('hover');
+  });
+  $('#side-dropzone').on('dragover',function(e) {
+    e.preventDefault();
+    $('#side-dropzone').addClass('hover');
+  });
+  $('.dropzone').on('dragout',function(e) {
+    $('.dropzone').removeClass('hover');
+  });
+  $('#side-dropzone').on('dragout',function(e) {
+    $('#side-dropzone').removeClass('hover');
+  });
+  $('.dropzone').on('drop',function(e) {
+    $D.selected = $(e.target.closest('div.comment-form-wrapper')).eq(0);
+    e.preventDefault();
+    // $E.initialize({});
```

Comment:
This is a big change (removing $E.initialize({})).

I tried to trace this in the history, it seems like this was originally done to address an image upload issue in /wiki/new... Ironically, I decided to comment this out now to fix other image upload issues in wikis. I think I'll see the consequences of this in the next few days as I play around with things, but so far it seems all right.

I chose this because I'm seeing that e.target is VERY unreliable in these click handlers (it's such a big issue I'm going to work on it next). I'd rather have $E be stable here, than have both $D and $E be reset with one event.

Reply:
> multiple handlers being set on multiple .dropzones, with bubbling

yikes 😱 😭 

but, thank you for starting to untangle this!!!

Reply:
OK, just before approving, is this resolved, or what is the latest status here? Would you like help determining what exact purpose `$E.initialize()` played here, that could help to inform or at least thoroughly document this removal? Thank you!!! Mostly looking to leave a good "paper trail" here as to why we choose to remove it.

Reply:
@jywarren Yes, if you have the time and willingness this week, would you mind looking into $E.initialize({}) here? I looked through the blame views, but I think it could help to have a second pair of eyes looking at this... I agree with you that we should be careful about the removal.

I'll take a deeper look again tonight investigating the consequences of removing $E.initialize... I tested it out with both `rails test` and just poking around about a week ago. I remember my impressions were that it was definitely a net positive, but I could write a more detailed comment about that.

#8927 is good to merge before this one! In fact, having that one merged might even help me out here, or maybe even sidestep this conversation.